src/host/pico_platform/include/pico/platform.h

```diff
@@ -135,5 +135,21 @@ static inline void __compiler_memory_barrier(void) {
 }
 #ifdef __cplusplus
 }
+
```

Comment:
note we should only need this for big endian systems... either omit, or make it a no-op on little endian host

Reply:
Sure.  A modern compiler will in fact already turn this code into a no-op on little endian, but I can put an explicitly no-op version inside an `#ifdef`.

Reply:
(Here's what I mean about the compiler fixing this for you, btw: https://godbolt.org/z/3cTdTYGzY)

Reply:
Yeah, I figured, I didn't know they all did so so uniformly... still it is clearer, and some maniacs may use -O0.